CamScanner Script

SmritiAgrawal04 opened the pull request:
# Description

A script to generate scanned copy of the original image is made.

Fixes #640

## Type of change

_Please delete options that are not relevant._

- [ ] New feature (non-breaking change which adds functionality)

# Checklist:

- [ ] My code follows the style guidelines(Clean Code) of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have created a helpful and easy to understand `README.md`
- [ ] My changes generate no new warnings

HarshCasper reviewed (changes requested):
Hi @SmritiAgrawal04 

Great Work! I have a few questions: Is the sole purpose of this Script to convert an Image to grayscale? 

If yes, can we think of some customizations to this, that can make it near to an actual camscanner?

rahulraikwar00 commented on Python/Camscanner/camscanner.py:
```diff
@@ -0,0 +1,19 @@
+	
+	# scannedImageFile= filename.split('.')[0]+'_scanned.jpeg'
+	# print (scannedImageFile)
+	# cv2.imwrite(scannedImageFile, scannedImage)
```
it is saving complete black png file !

rahulraikwar00 reviewed (commented):
Unable to exit after running script! It gets stuck on the running script !

SmritiAgrawal04 commented:
No it also works to enhance the image as is done in a CamScanner. Yet I'll
look into it again and see what best can be done.😁

On Wed, Mar 31, 2021, 2:46 PM Harsh Mishra ***@***.***> wrote:

> *@HarshCasper* requested changes on this pull request.
>
> Hi @SmritiAgrawal04 <https://github.com/SmritiAgrawal04>
>
> Great Work! I have a few questions: Is the sole purpose of this Script to
> convert an Image to grayscale?
>
> If yes, can we think of some customizations to this, that can make it near
> to an actual camscanner?
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/HarshCasper/Rotten-Scripts/pull/918#pullrequestreview-625069312>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ANCEMHZVGC2OFBX4LMAFIDDTGLR55ANCNFSM42CNIBBQ>
> .
>

SmritiAgrawal04 commented:
Hi @vybhav72954 

I've changed the script and added 8 ways to scan the image, please check if any of these suffices our requirements. The script runs the same way as before- check the README. Also, @rahulraikwar00 the script now does not get stuck.

vybhav72954 commented on Python/Camscanner/README.md:
```diff
@@ -0,0 +1,15 @@
+# Generate Scanned Image
```
Hey @SmritiAgrawal04 Can we please have a more detailed README 😅

SmritiAgrawal04 replied:
Done! Can you please check if this looks good now?

vybhav72954 reviewed (changes requested):
@HarshCasper After this one change.

Looks good to me.

rahulraikwar00 commented on Python/Camscanner/README.md:
```diff
@@ -0,0 +1,26 @@
+    
+    
+## References:
+1. https://opencv-python-tutroals.readthedocs.io/en/latest/py_tutorials/py_imgproc/py_thresholding/py_thresholding.html
```
@SmritiAgrawal04 throwing error when clicked on the attachment Hyperlink 
Error - Non-Image content-type returned
quick fix: just remove the exclamation mark from front

rahulraikwar00 reviewed (changes requested):
and everything LGTM 👍

rahulraikwar00 reviewed (approved):
Looks fine..!

SmritiAgrawal04 commented:
can we merge this pull request then?

SmritiAgrawal04 commented:
> can we merge this pull request then?

Hey @vybhav72954 !
I request you to please merge this pull request.

HarshCasper commented on Python/Camscanner/README.md:
```diff
@@ -0,0 +1,26 @@
+    
+    
+## References:
+1. https://opencv-python-tutroals.readthedocs.io/en/latest/py_tutorials/py_imgproc/py_thresholding/py_thresholding.html
+
+2. https://medium.com/@anupriyam/basic-image-thresholding-in-opencv-5af9020f2472
```
Add them as hyperlinks, not plain links

HarshCasper commented on Python/Camscanner/README.md:
```diff
@@ -0,0 +1,26 @@
+## To run the script:
+1. Install the dependencies by running following command in terminal.
+
+   `pip install opencv-python`
+   `pip install numpy`
+   `pip install matplotlib`
```
Add a `requirements.txt` file and add all of these dependencies there
